gensim/models/ldaseqmodel.py

```diff
@@ -0,0 +1,1222 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+# Based on Copyright (C) 2016 Radim Rehurek <radimrehurek@seznam.cz>
+
+
+"""
+
+This is the class which is used to help with Dynamic Topic Modelling of a corpus.
+Inspired by the Blei's original DTM code and paper. 
+DTM C/C++ code: https://github.com/blei-lab/dtm
+DTM Paper: https://www.cs.princeton.edu/~blei/papers/BleiLafferty2006a.pdf
+
+
+"""
+
+from gensim import interfaces, utils, matutils
+from gensim.models import ldamodel
+import numpy
+import math
+from scipy.special import digamma
+from scipy import optimize
+
+
+class seq_corpus(utils.SaveLoad):
+
+    """
+    `seq_corpus` is basically a wrapper class which contains information about the corpus.
+    `vocab_len` is the length of the vocabulary.
+    `max_doc_len` is the maximum number of terms a single document has.
+    `num_time_slices` is the number of sequences, i.e number of time-slices.
+    `corpus_len` is the number of documents present.
+    `time_slice` is a list or numpy array which the user must provide which contains the number of documents in each time-slice.
+    `corpus` is any iterable gensim corpus.
+
+    """
+    def __init__(self, corpus=None, time_slice=None, id2word=None):
+
+
+        self.id2word = id2word
+        if corpus is None and self.id2word is None:
+            raise ValueError('at least one of corpus/id2word must be specified, to establish input space dimensionality')
+
+        if self.id2word is None:
+            logger.warning("no word id mapping provided; initializing from corpus, assuming identity")
+            self.id2word = utils.dict_from_corpus(corpus)
+            self.vocab_len = len(self.id2word)
+        elif len(self.id2word) > 0:
+            self.vocab_len = len(self.id2word)
+        else:
+            self.vocab_len = 0
+
+        self.corpus = corpus
+        if self.corpus is not None:
+            self.corpus_len = len(corpus)
+
+        self.time_slice = time_slice
+        if self.time_slice is not None:
+            self.num_time_slices = len(time_slice)
+
+        max_doc_len = 0
+        for line_no, line in enumerate(corpus):
+            if len(line) > max_doc_len:
+                max_doc_len = len(line)
+        self.max_doc_len = max_doc_len
+
+
+class Doc(utils.SaveLoad):
+    """
+    The doc class contains information used for each document. 
+
+    """
+    def __init__(self, nterms=None, word=None, count=None, total=None):
+
+        self.nterms = nterms
+        self.word = word
+        self.count = count
+        self.total = total
+
+
+class LdaSeqModel(utils.SaveLoad):
+    """
```

Comment:
Add how to use similar to https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/ldamodel.py#L200